core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2954,6 +2954,15 @@ private[spark] object Utils
     str.replaceAll("[ :/]", "-").replaceAll("[.${}'\"]", "_").toLowerCase(Locale.ROOT)
   }
 
+  def nameForAppAndAttempt(appId: String, appAttemptId: Option[String]): String = {
```

Comment:
this can be reused in several places, for example https://github.com/apache/spark/pull/42575#discussion_r1299557854